crates/trigger-http2/src/handler.rs

```diff
@@ -0,0 +1,411 @@
+use std::{net::SocketAddr, str, str::FromStr};
+
+use anyhow::{anyhow, Context, Result};
+use futures::TryFutureExt;
+use http::{HeaderName, HeaderValue};
+use http_body_util::BodyExt;
+use hyper::{Request, Response};
+use spin_core::{Component, Instance};
+use spin_factor_outbound_http::wasi_2023_10_18::exports::wasi::http::incoming_handler::Guest as IncomingHandler2023_10_18;
+use spin_factor_outbound_http::wasi_2023_11_10::exports::wasi::http::incoming_handler::Guest as IncomingHandler2023_11_10;
+use spin_http::body;
+use spin_http::routes::RouteMatch;
+use spin_world::v1::http_types;
+use tokio::{sync::oneshot, task};
+use tracing::{instrument, Instrument, Level};
+use wasmtime_wasi_http::{body::HyperIncomingBody as Body, proxy::Proxy, WasiHttpView};
+
+use crate::{server::HttpExecutor, Store, TriggerInstanceBuilder};
+
+#[derive(Clone)]
+pub struct HttpHandlerExecutor {
```

Comment:
It'd be good to document what this is - an `HttpExector` that can execute wasi and spin http incoming requests.